projects/kit/components/slider/slider.component.ts

```diff
@@ -60,6 +68,29 @@ export class TuiSliderComponent {
         }
     }
 
+    // TODO(v5): use signal inputs
+    @Input({
+        alias: 'segments',
+        transform: (segments: number[] | number) => {
+            if (Array.isArray(segments)) {
+                const arr = segments.map((segment) => segment * 100);
+
+                if (arr[0] !== 0) {
+                    arr.unshift(0);
+                }
+
+                return arr;
+            }
+
+            return new Array(segments)
+                .fill(null)
+                .map((_, index) => (index * 100) / segments);
```

Comment:
I am not sure that `@Input`'s transform should be so huge 🤔 

Let's just transform single `number` to `number[]` (with ratios, not percentages).
```ts
@Input({
    alias: 'segments',
    transform: (x: number[] | number) =>
        Array.isArray(x) ? x : new Array(x).fill(null).map((_, i) => i / x),
})
public set segmentsSetter(segments: number[]) {
```